app/src/main/java/com/imashnake/animite/features/MainActivity.kt

```diff
@@ -39,65 +44,72 @@ class MainActivity : ComponentActivity() {
-                    navController = navController
-                )
+                    val bottomPadding: Dp by animateDpAsState(
+                        if (
+                            NavigationBarPaths.values().map { it.direction }.any {
+                                navController.appCurrentDestinationAsState().value == it
+                            }
+                        ) bottomNavigationBarSize else 0.dp
+                    )
 
-                AnimatedVisibility(
-                    visible = NavigationBarPaths.values().map { it.direction }.any {
-                        navController.appCurrentDestinationAsState().value == it
-                    },
-                    modifier = Modifier
-                        .align(Alignment.BottomCenter)
-                        .height(
-                            80.dp + WindowInsets
-                                .navigationBars
-                                .asPaddingValues()
-                                .calculateBottomPadding()
-                        ),
-                    enter = slideInVertically { it },
-                    exit = slideOutVertically { it }
-                ) {
-                    NavigationBar(
-                        navController = navController,
+                    SearchBar(
+                        modifier = Modifier
+                            .align(Alignment.BottomEnd)
+                            .padding(
+                                start = 24.dp,
+                                end = 24.dp,
+                                bottom = bottomPadding
+                            )
+                            .navigationBarsPadding()
+                            .padding(bottom = 24.dp),
+                        navController = navController
+                    )
+
+                    AnimatedVisibility(
+                        visible = NavigationBarPaths.values().map { it.direction }.any {
+                            navController.appCurrentDestinationAsState().value == it
+                        },
                         modifier = Modifier
                             .align(Alignment.BottomCenter)
                             .height(
-                                80.dp + WindowInsets
+                                bottomNavigationBarSize + WindowInsets
                                     .navigationBars
                                     .asPaddingValues()
                                     .calculateBottomPadding()
                             ),
-                        itemModifier = Modifier.navigationBarsPadding()
-                    )
+                        enter = slideInVertically { it },
+                        exit = slideOutVertically { it }
+                    ) {
+                        NavigationBar(
+                            navController = navController,
+                            modifier = Modifier.align(Alignment.BottomCenter),
```

Comment:
`BottomCenter` shouldn't be necessary here, the parent is already aligned correctly. Consider moving `height` to the NavigationBar Composable and letting AnimatedVisibility wrap content